src/lib.rs

```diff
@@ -424,42 +423,37 @@ It permits resilient arbitrary dynamic membership changes. A membership change c
 the following:
 
 * Add peer (learner or voter) *n* to the group.
-* Remove peer *n* from the group.
-* Remove a leader (unmanaged, via stepdown)
+* Remove a learner *n* from the group.
 * Promote a learner to a voter.
+* Demote a voter back to learner.
 * Replace a node *n* with another node *m*.
 
-It (currently) does not:
-
-* Allow control of the replacement leader during a stepdown.
-* Optionally roll back a change during a peer group pause where the new peer group configuration
-fails.
-* Provide automated promotion of newly added voters from learner to voter when they are caught up.
-This must be done as a two stage process for now.
-
-> PRs to enable these are welcome! We'd love to mentor/support you through implementing it.
-
-This means it's possible to do:
-
-```rust
-use raft::{Config, storage::MemStorage, raw_node::RawNode, eraftpb::*};
-use protobuf::Message as PbMessage;
-use slog::{Drain, o};
-
-let mut config = Config { id: 1, ..Default::default() };
-let store = MemStorage::new_with_conf_state((vec![1, 2], vec![]));
-let logger = slog::Logger::root(slog_stdlog::StdLog.fuse(), o!());
-let mut node = RawNode::new(&mut config, store, &logger).unwrap();
-node.raft.become_candidate();
-node.raft.become_leader();
-
+For example to promote a learner 4 and demote an existing voter 3:
```

Comment:
Seems 3 and 4 is not in config?

Reply:
It's just a code showing how to construct joint changes, it doesn't matter whether nodes are in configurations as long as it can compile.

Reply:
Codes after `#` will not be decorated.